src/services/transaction_watcher_service.ts

```diff
@@ -0,0 +1,116 @@
+import { SupportedProvider, Web3Wrapper } from '@0x/web3-wrapper';
+import { Connection, Not, Repository } from 'typeorm';
+
+import { TX_WATCHER_POLLING_INTERVAL_MS } from '../constants';
+import { TransactionEntity } from '../entities';
+import { logger } from '../logger';
+import { TransactionStates } from '../types';
+import { utils } from '../utils/utils';
+
+export class TransactionWatcherService {
+    private readonly _transactionRepository: Repository<TransactionEntity>;
+    private readonly _provider: SupportedProvider;
+    private readonly _web3Wrapper: Web3Wrapper;
+
+    constructor(dbConnection: Connection, provider: SupportedProvider) {
+        this._transactionRepository = dbConnection.getRepository(TransactionEntity);
+        this._provider = provider;
+        this._web3Wrapper = new Web3Wrapper(this._provider);
+    }
+    public async startAsync(): Promise<void> {
+        while (true) {
```

Comment:
We comment about using `intervalUtils.setAsyncExcludingInterval` here instead, so we can tear down this job.